sktime/forecasting/compose/_pipeline.py

```diff
@@ -1264,6 +1264,10 @@ def _get_forecaster_X_prediction(self, X=None, fh=None, method="predict"):
         if X is not None:
             X_pred = X_pred.combine_first(X)
```

Comment:
I think here is another issue, its assuming that the specified columns are `NaN`, if not it will not overwrite the values with the predictions. So this is then making issues in `evalute` and therefore in the grid searches as well. Happy to make a PR if you dont want to fix it? Its a separate bug of `ForecastX` I think?

Reply:
sure, although I'm not sure what the bug is or what the condition is under which it appears

Reply:
I remember checking which way round it is, and I do think it prioritizes observations over predictions. I think that's the right way round? Since observations are more "real" than predictions.

Reply:
I don't think this is a problem of `ForecastX` because it's expected behaviour. Instead, `evaluate` should censor the variables that are not known in advance when feeding them to the estimator.

Reply:
We could have a param to handle this in `ForecastX` or change the logic (my preferred solution), currently its not possible to backtest this and therefore its not possible to test this in grid searches as it will always use the real observations if I am not wrong?

Reply:
> We could have a param to handle this in

why not - the default should be "observation overwrites forecast" imo as that is user expectation.

However, guessing what is your train of thought: `evaluate` should not know about the estimators it is getting, or you get a highly coupled design.

To address your problem, I would instead suggest to add a "censored variables" or "temporal variables" option and use that in preparing your `X` and `y` in `evaluate`.